linera-client/src/chain_listener.rs

```diff
@@ -61,17 +60,16 @@ pub struct ChainListenerConfig {
 
 type ContextChainClient<C> = ChainClient<<C as ClientContext>::Environment>;
 
-#[cfg_attr(not(web), async_trait, trait_variant::make(Send))]
-#[cfg_attr(web, async_trait(?Send))]
-pub trait ClientContext: 'static {
+#[cfg_attr(not(web), trait_variant::make(Send))]
+#[allow(async_fn_in_trait)]
+pub trait ClientContext {
```

Comment:
If we could extract `update_wallet_for_new_chain` and `update_wallet` - this method wouldn't have any async fns anymore.

Reply:
I plan to completely revamp this trait in a short while after moving the wallet into the client, so I'm not keen to do it here.